### PR TITLE
Add scheduler pallet

### LINKED ISSUE #289: [Governance] Add and enable scheduler pallet
Scheduler pallet is needed for upgrades to target a specific block.

### PR DESCRIPTION
**Pull request by shannonwells**
# Goal
So democracy features such as referenda, upgrades and other extrinsics can be set to execute at a future block, add and configure the scheduler pallet.

Closes #289 

# Discussion

# Checklist
- [ ] Chain spec updated
- [x] <del>Updated the js/api-augment code if a custom RPC added/changed</del> Not applicable
- [x] <del>Design doc(s) updated</del> Not applicable
- [x] <del>Tests added</del> Not applicable
- [x] <del>Benchmarks added</del> Not applicable
- [ ] Weights updated

### CODE REVIEW

**Comment by enddynayn** [comment]
👍 looking good I think you just need to add weights.

File: runtime/frequency/src/lib.rs [hunk 1]
```diff
@@ -373,6 +373,28 @@ impl pallet_balances::Config for Runtime {
...
+	type Event = Event;
+	type Origin = Origin;
+	type PalletsOrigin = OriginCaller;
+	type Call = Call;
+	type MaximumWeight = MaximumSchedulerWeight;
+	type ScheduleOrigin = frame_system::EnsureRoot<AccountId>;
+	type MaxScheduledPerBlock = MaxScheduledPerBlock;
+	type WeightInfo = ();
+	type OriginPrivilegeCmp = EqualPrivilegeOnly;
+	type PreimageProvider = ();
```
Comment (enddynayn) [inline runtime/frequency/src/lib.rs:394]: Curious, why not use `pallet_preimage` to store the hash of the image?

Comment (shannonwells) [reply runtime/frequency/src/lib.rs:394]: Ah thanks, I was debugging some stuff and I forgot to put the Preimage config back.

File: runtime/frequency/src/lib.rs [hunk 2]
```diff
@@ -373,6 +373,43 @@ impl pallet_balances::Config for Runtime {
...
+}
+
+parameter_types! {
+	pub const PreimageMaxSize: u32 = 4096 * 1024;
+	pub const PreimageBaseDeposit: Balance = 1 * MILLIUNIT;
+	pub const PreimageByteDeposit: Balance = 1 * MICROUNIT;
+}
+
+impl pallet_preimage::Config for Runtime {
+	type WeightInfo = pallet_preimage::weights::SubstrateWeight<Runtime>;
```
Comment (enddynayn) [inline runtime/frequency/src/lib.rs:405]: Maybe I am wrong but I think we need to generate these weights right?

Comment (enddynayn) [reply runtime/frequency/src/lib.rs:405]: I think I found our answer on [stackexchange](https://substrate.stackexchange.com/questions/295/how-to-generate-weightinfo-for-pallets).

File: runtime/frequency/src/lib.rs [hunk 3]
```diff
@@ -373,6 +373,43 @@ impl pallet_balances::Config for Runtime {
 	type ReserveIdentifier = [u8; 8];
 }
 
+parameter_types! {
+	pub MaximumSchedulerWeight: Weight = Perbill::from_percent(10) * RuntimeBlockWeights::get().max_block;
```
Comment (aramikm) [inline runtime/frequency/src/lib.rs:377]: Would you please add some comments about what is this weight for and how it's been used?

**Review by aramikm — COMMENTED** [review]
I'm also curious about how can we generate our own weights for included pallets. Other than that looks good.

File: runtime/frequency/src/lib.rs (hunk 2 above)
Comment (shannonwells) [reply runtime/frequency/src/lib.rs:407]: Added scheduler to benchmarks + weights file.

**Review by enddynayn — APPROVED** [review]

**Pull request merged**